zerver/tests/test_signup.py

```diff
@@ -1171,9 +1171,8 @@ def test_digest_unsubscribe(self) -> None:
 
         # The setting is toggled off, and scheduled jobs have been removed.
         self.assertEqual(result.status_code, 200)
-        # Circumvent user_profile caching.
 
-        user_profile.refresh_from_db()
+        user_profile = self.example_user("hamlet")
         self.assertFalse(user_profile.enable_digest_emails)
```

Comment:
I think the intent here is to make sure that we didn't incorrectly only update the cached `user_profile` (or a local in-memory object, even!), but in fact actually wrote a change through to the database.

Reply:
I just don't think that was the intent of the test.  I think you're overthinking this.  They probably just cargo-culted the `refresh_from_db` test.  They certainly didn't put enough thought into the test to either a) just write a comment or b) just go directly to the database to see what was written.

Reply:
From my experiences of adding `refresh_from_db` calls in tests, I think it was done because the setting value for user/realm defined before changing the setting do not have updated setting values, unless the action performed to change the setting returned the new object itself, which does not happen when we make API requests to change the settings. 

And I think we do not return the user from `memcached` when using `example_user` since it uses [`get_user_by_delivery_email`](https://github.com/zulip/zulip/blob/4300ec329330869b69ba17d94e057df0b7351517/zerver/models.py#L3826).